### PR TITLE
Add severites

### PR DESCRIPTION
**Pull request by joshuai96**
- Add severities to database
- Add severities as resource to API
- Add severities to impacts (WIP)

### CODE REVIEW

**Comment by joshuai96** [comment]
@snkaupe FYI

**Comment by joshuai96** [comment]
Zuul recheck

**Comment by joshuai96** [comment]
@master-caster does recheck not work?

File: docs/requests.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+As defined by the [OpenAPI spec](https://github.com/SovereignCloudStack/status-page-openapi) and [status page OpenAPI decision](https://github.com/SovereignCloudStack/standards/blob/main/Standards/scs-0402-v1-status-page-openapi-spec-decision.md) the general API objects are used as request bodies and respponses to generalize data structures. Not all object fields are handled by all requests, some are read only and some are write only. `GET` request wrap their return in a object field called `data`.
+
+Please refer to [example requests](example-requests.md) to see these request in action.
+
+## Phases
+
+Phases are always handled as lists, so `GET` as well as `POST` operations on phases always require the full list. When geting the phase list, it's accompanied be a generation annotation.
+
+```json
```
Comment (joshmue) [inline docs/requests.md:11]:
```suggestion
```json5
```

Makes the comments render "correctly".

File: docs/requests.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    "Pahse 3"
+  ]
+}
+```
+
+## Impact types
+
+Requesting (`GET`) an impact type, will return all fields, while `POST` and `PATCH` operations ommit the `id` field.
+
+```json
```
Comment (joshmue) [inline docs/requests.md:26]:
```suggestion
```json5
```

File: docs/requests.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+}
+```
+
+As `displayName` is the identifier it must be unique, even when modified by `PATCH`
+
+## Components
+
+When `GET`ing a component, all fields can be expected to be filled, while requests for `POST` (creation) and `PATCH` operations only handle certain fields.
+
+```json
```
Comment (joshmue) [inline docs/requests.md:51]:
```suggestion
```json5
```

File: docs/requests.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    "key": "value",
+  }
+}
+```
+
+## Incidents
+
+It is expected that incidents are the most used API object and have the most data to transmit.
+
+```json
```
Comment (joshmue) [inline docs/requests.md:72]:
```suggestion
```json5
```

File: docs/requests.md [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+The `affects` field can, in theory, have as many entries as there are components. The `updates` field is an ongoing list of updates. All changes to `phase` should correlate to an update.
+
+When performing `POST` or `PATCH` operations on incidents the `affects` field is of utmost importance, as it creates the **impact**. Only when referencing a component to an incident via the `affects` field, an impact is created, that can be retrieved via the affected component.
+
+## Incident update
+
+Whenever an incident changes, an update should be issued. When doing a `GET` request, the `order` field is filled, updates should be displayed in ascending order.
+
+```json
```
Comment (joshmue) [inline docs/requests.md:106]:
```suggestion
```json5
```

File: pkg/db/db.go [hunk 6]
```diff
@@ -34,14 +36,76 @@ func (m *Model) BeforeCreate(_ *gorm.DB) error {
...
 }
 
+func provision[S ~[]E, E any](data S, dbTx *gorm.DB, logger *zerolog.Logger) error {
+	var (
+		limit  = 5
+		target = make(S, limit)
+	)
+
+	typ := reflect.TypeOf(data).Elem()
+	provisioningLogger := logger.With().Str("function", "provision").Str("type", typ.Name()).Logger()
```
Comment (joshmue) [inline pkg/db/db.go:46]:
```suggestion
	reflectType := reflect.TypeOf(data).Elem()
	provisioningLogger := logger.With().Str("function", "provision").Str("type", reflectType.Name()).Logger()
```

File: pkg/db/db.go [hunk 7]
```diff
@@ -34,14 +36,76 @@ func (m *Model) BeforeCreate(_ *gorm.DB) error {
...
+
+	// get from database if exist.
+	res := dbTx.
+		Limit(limit). // do not load all.
+		Find(&target)
+	if res.Error != nil {
+		if !errors.Is(res.Error, gorm.ErrRecordNotFound) {
+			return fmt.Errorf("error getting %ss: %w", typ.Name(), res.Error)
```
Comment (joshmue) [inline pkg/db/db.go:54]:
```suggestion
			return fmt.Errorf("error getting %s: %w", typ.Name(), res.Error)
```

**Review by joshmue — APPROVED** [review]
LGTM overall.

**Pull request merged**